src/hermez-wallet.js

```diff
@@ -101,7 +102,7 @@ class HermezWallet {
 async function createWalletFromEtherAccount (providerUrl, signerData) {
   const provider = getProvider(providerUrl)
   const signer = getSigner(provider, signerData)
-  const ethereumAddress = await signer.getAddress(signerData && signerData.addressOrIndex)
+  const ethereumAddress = await signer.getAddress()
```

Comment:
`getSigner` does not receive any input parameter:
https://docs.ethers.io/v5/api/signer/#Signer-getaddress
If `signerData.addressOrIndex` is present on `signerData` input parameter it will be loaded in previous function `getSigner` where JSON-RPC is loaded with addressOrIndex such as `provider.getSigner(signerData.addressOrIndex)`.
Therefore, `signer.getAddress()` would return the public address not depending on `signerData.addressOrIndex` but depending on `getSigner`